src/main/java/pl/most/typer/controller/footballController/MatchController.java

```diff
@@ -48,39 +52,56 @@ public String updateMatchInfo(@PathVariable(value = "id") Integer id) {
     private String getMatchesByStages(@PathVariable("id") Integer id,
                                       @RequestParam(name = "stage", required = false) String stage,
                                       @RequestParam(name = "TCID", required = false) String typerCompetitionId,
-                                      Model model) {
+                                      Model model) throws ResourceException {
         Optional<Competition> optionalCompetition = Optional.of(competitionService.getCompetition(id));
         Optional<Match> optionalMatch = matchesService.findFirstByCompetition(new Competition(id, null));
 
         if (checkIfCompetitionMatchExist(optionalCompetition, optionalMatch)) {
             return "redirect:/competitions/" + id + "/matches/update";
         }
 
-        List<Match> matchByCompetition;
+        List<Match> matchByCompetition = null;
 
         if (stage != null) {
-            matchByCompetition = matchesService
-                    .findAllByCompetitionAndStage(optionalCompetition.get(), stage);
+            try {
+                matchByCompetition = matchesService
+                        .findAllByCompetitionAndStage(optionalCompetition.get(), stage);
+            } catch (ResourceException ex) {
+                log.warn(ex.getMessage());
+                throw ex;
+            }
         } else {
-            List<String> stages = matchesService.getStages(optionalCompetition.get());
-            if (stages.size() == 1) {
-                return "redirect:/competitions/" + id + "/matches?stage=" + stages.stream().findFirst().get();
+            try {
+                List<String> stages = matchesService.getStages(optionalCompetition.get());
+                if (stages.size() == 1) {
+                    return "redirect:/competitions/" + id + "/matches?stage=" + stages.stream().findFirst().get();
+                }
+
+                model.addAttribute("apiId", id);
+                model.addAttribute("competitionName", competitionService.getCompetitionName(id));
+                model.addAttribute("stages", stages);
+
+                return "stages";
+            } catch (ResourceException e) {
+                log.warn(e.getMessage());
+                throw e;
             }
 
-            model.addAttribute("apiId", id);
-            model.addAttribute("competitionName", competitionService.getCompetitionName(id));
-            model.addAttribute("stages", stages);
-
-            return "stages";
         }
 
-        LinkedHashMap<String, List<Match>> matchesByGroup = getGroupLinkedHashMap(matchByCompetition);
-
-        model.addAttribute("competitionName", competitionService.getCompetitionName(id));
-        model.addAttribute("matchesByGroup", matchesByGroup);
-        model.addAttribute("stageName", stage);
-        model.addAttribute("TCID",typerCompetitionId);
-
+        if (matchByCompetition != null) {
```

Comment:
moznaby to przeniesc do 69 linijki kodu i wtedy nie musisz sprawdzac nulla, a wywolanie tego jest potrzebne tylko wtedy bo w innych przypadkach juz wczesniel byl return. Chodzi o calego ifa razem z return